library/model_support.c

```diff
@@ -381,6 +439,66 @@ static int _parse_timeval(timestamp *t, const char *json, jsmntok_t *tok) {
     return 1;
 }
 
+static int _cmp_bool(bool *lh, bool *rh) {
+    if (lh == rh) { return 0; }
+    if (lh == NULL) { return -1; }
+    if (rh == NULL) { return 1; }
+    if (*lh == *rh) { return 0; }
+    if (!*lh) { return -1; }
+    return 1;
+}
+
+static int _cmp_int(int *lh, int *rh) {
+    if (lh == rh) { return 0; }
+    if (lh == NULL) { return -1; }
+    if (rh == NULL) { return 1; }
+
```

Comment:
Surprised you didn't make a macros for the lh/rh NULL checks :)

Reply:
good idea Dave!